site/docs/build-event-protocol.md

```diff
@@ -0,0 +1,152 @@
+---
+layout: documentation
+title: Build Event Protocol
+---
+
+# Build Event Protocol
+
+The [Build Event Protocol] allows third party programs to gain insight into a Bazel invocation. The protocol is a set of [protocol buffer] messages with some semantics defined on top of it. It includes information about build and test results, build progress, the build configuration and much more. The Build Event Protocol is intended to be consumed programmatically and makes parsing Bazel’s command line output a thing of the past. For example, using the information from the Build Event Protocol one could create a dashboard that for a given build displays the targets built, tests run and artifacts generated. Another example would be an IDE plugin gathering information about the build targets and tests being executed.
+
+## The Build Event Protocol from 10,000 feet
+As mentioned before, the [Build Event Protocol] represents information about a build as events. A build event is a protocol buffer message consisting of a build event identifier, a set of child events and a payload.
+ 
+* __Build Event Identifier:__
+    Depending on the kind of build event, it might be an [opaque string] or [structured information] revealing more about the build event. A build event identifier is unique within a build.
+
+* __Children:__ A build event may announce other build events, by including their build event identifiers in its [children field]. For example, the `PatternExpanded` build event announces the targets it expands to as children. The main purpose of child events is to ensure that no event except for the initial event is posted unexpectedly.
+
+* __Payload:__ The payload contains structured information about a build event, encoded as a protocol buffer message specific to that event.
+
+### Build Event Graph
+All build events form a [directed acyclic graph] through their parent and child relationship. Every build event except for the initial build event has one or more parent events. Please note that not all parent events of a child event must necessarily be posted before it. When a build is complete (succeeded or failed) all announced events will have been posted. In case of errors i.e. Bazel crashed or a failed network transport, some announced build events may never be posted.
```

Comment:
In fact, we even have some events guaranteed to come *after* their children: the action complete event for a failed action that is root cause of target failure will always come before the targetComplete event that refers to it (as a child event)